ziti/cmd/edge/create_extjwtsigner.go

```diff
@@ -0,0 +1,161 @@
+			options.Cmd = cmd
+			options.Args = args
+			err := runCreateExtJwtSigner(options)
+
+			cmdhelper.CheckErr(err)
+		},
+		SuggestFor: []string{},
+	}
+
+	// allow interspersing positional args and flags
+	cmd.Flags().SetInterspersed(true)
+	cmd.Flags().BoolVarP(options.ExtJwtSigner.UseExternalID, "external-id", "x", true, "Matches identity external ids rather than internal ids")
+	cmd.Flags().BoolVarP(options.ExtJwtSigner.Enabled, "enabled", "e", true, "Enable this entity")
+	cmd.Flags().StringVarP(&options.JwksEndpoint, "jwks-endpoint", "u", "", "A valid URI for a target JWKS endpoint, not usable with -p or -f")
+	cmd.Flags().StringVarP(options.ExtJwtSigner.ClaimsProperty, "claims-property", "c", "sub", "The JWT property matched to identity internal/external ids")
+	cmd.Flags().StringVarP(options.ExtJwtSigner.Audience, "audience", "a", "", "The expected audience of the incoming JWTs")
+	cmd.Flags().StringVarP(options.ExtJwtSigner.CertPem, "cert-pem", "p", "", "A public certificate PEM for the signer, not usable with -u, -f")
+	cmd.Flags().StringVarP(&options.CertFilePath, "cert-file", "f", "", "A public certificate PEM file, not usable with -u, -p")
+	cmd.Flags().StringVarP(options.ExtJwtSigner.ExternalAuthURL, "external-auth-url", "y", "", "The URL that users are directed to obtain a JWT")
+	cmd.Flags().StringVarP(options.ExtJwtSigner.Kid, "kid", "k", "", "The KID for the signer, required if using -p or -f")
+	options.AddCommonFlags(cmd)
+
+	return cmd
+}
+
+func runCreateExtJwtSigner(options *createExtJwtSignerOptions) (err error) {
+	managementClient, err := util.NewEdgeManagementClient(options)
+
+	if err != nil {
+		return err
+	}
+
+	hasJwks := options.JwksEndpoint != ""
+	hasCertPem := options.ExtJwtSigner.CertPem != nil || *options.ExtJwtSigner.CertPem == ""
+	hasCertPath := options.CertFilePath != ""
+
+	if (hasJwks && (hasCertPem || hasCertPath)) || (hasCertPem && (hasJwks || hasCertPath)) || (hasCertPath && (hasJwks || hasCertPem)) {
+		return errors.New("must specify only one certificate source (JWKS, cert file, or inline cert PEM")
+	}
+
+	if hasCertPem || hasCertPath && (options.ExtJwtSigner.Kid == nil || *options.ExtJwtSigner.Kid == "") {
+		return errors.New("must specify a KID if using a certificate file or inline PEM")
+	}
+
+	if options.ExtJwtSigner.Audience == nil || *options.ExtJwtSigner.Audience == "" {
+		return errors.New("audience must be specified")
+	}
+
+	if options.ExtJwtSigner.ClaimsProperty != nil && *options.ExtJwtSigner.ClaimsProperty == "" {
+		return errors.New("claims property must not be an empty string")
+	}
+
+	if options.JwksEndpoint != "" {
+		jwks := strfmt.URI(options.JwksEndpoint)
+		options.ExtJwtSigner.JwksEndpoint = &jwks
+	}
+
+	if hasCertPath {
+		pem, err := os.ReadFile(options.CertFilePath)
+		if err != nil {
+			return fmt.Errorf("could not read certificate file path %s: %w", options.CertFilePath, err)
+		}
+		pemStr := string(pem)
+		options.ExtJwtSigner.CertPem = &pemStr
```

Comment:
if the file is empty or simply "not a pem" i hope that it's made clear somehwere down the line that the cert is invalid?

Reply:
The API will parse it and verify it works. I could do it client-side too, but I don't feel strongly either way.